server/src/main/java/io/crate/types/ObjectType.java

```diff
@@ -383,11 +383,25 @@ public StorageSupport<Map<String, Object>> storageSupport() {
     }
 
     @Override
-    public long valueBytes(Map<String, Object> value) {
-        if (value == null) {
+    @SuppressWarnings("unchecked")
+    public long valueBytes(Map<String, Object> map) {
+        if (map == null) {
             return RamUsageEstimator.NUM_BYTES_OBJECT_HEADER;
```

Comment:
Do we need such change in other types using Lucene's estiamtor? GeoShape/UnchecedType

Reply:
I think geo shape should use a different approach given that its value structure is closed with known properties (type, coordinates, geometries, etc.)

Doing type-guessing on the entries would be strange.

uncheckedType: As far as I can tell it's currently not used that way, but wouldn't hurt.